lib/private/handlers/files.ts

```diff
@@ -61,21 +62,30 @@ export class FileAssetHandler implements IAssetHandler {
 
     const bucketInfo = BucketInformation.for(this.host);
 
-    // A thunk for describing the current account. Used when we need to format an error
-    // message, not in the success case.
-    const account = async () =>
-      (await this.host.aws.discoverTargetAccount(clientOptions))?.accountId;
-    switch (await bucketInfo.bucketOwnership(s3, destination.bucketName)) {
+    const account = (await this.host.aws.discoverTargetAccount(clientOptions)).accountId;
+    switch (await bucketInfo.bucketOwnership(s3, destination.bucketName, account)) {
       case BucketOwnership.MINE:
         break;
       case BucketOwnership.DOES_NOT_EXIST:
         throw new Error(
-          `No bucket named '${destination.bucketName}'. Is account ${await account()} bootstrapped?`
+          `No bucket named '${destination.bucketName}'. Is account ${account} bootstrapped?`
         );
-      case BucketOwnership.SOMEONE_ELSES_OR_NO_ACCESS:
+      case BucketOwnership.NO_ACCESS:
         throw new Error(
-          `Bucket named '${destination.bucketName}' exists, but not in account ${await account()}. Wrong account?`
+          `Bucket named '${destination.bucketName}' exists, but we have no access to it.`
         );
+      case BucketOwnership.SOMEONE_ELSES:
+        if (destination.bucketName.includes(`-${account}-`)) {
+          // this is highly irregular. the bucket name suggests the bucket should belong
+          // to the target account, but it actually belongs to a different account.
+          throw new Error(
+            `Bucket named '${destination.bucketName}' exists, but not in account ${account}. Wrong account?`
+          );
+        } else {
+          verbose(
+            `Bucket named ${destination.bucketName} exists, but not in account ${account}. Assuming cross account setup and proceeding.`
```

Comment:
We could throw here as well - but there's a good chance this is an intentional cross account setup.

Reply:
Outdated